pkg/apis/fake/v1alpha1/db_types.go

```diff
@@ -0,0 +1,274 @@
+/*
+Copyright 2019 The KubeCarrier Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DBSpec defines the desired state of DB
+type DBSpec struct {
+	// RootPassword is root account password for this data. Leave blank for auto-generation
+	RootPassword string `json:"rootPassword"`
+
+	// DatabaseName of the created database at start up
+	DatabaseName string `json:"databaseName"`
+
+	// DatabaseUser for created database
+	DatabaseUser string `json:"databaseUser"`
+
+	// DatabasePassword for the created database. Leave blank for auto-generation
+	DatabasePassword string `json:"databasePassword"`
+
+	// E2E tests params
+	Config Config `json:"config,omitempty"`
```

Comment:
Why do you have Config under a spec....why not simply having those two fields inside the spec?